src/librustc_ast_lowering/expr.rs

```diff
@@ -483,14 +484,44 @@ impl<'hir> LoweringContext<'_, 'hir> {
             Some(ty) => FunctionRetTy::Ty(ty),
             None => FunctionRetTy::Default(span),
         };
-        let ast_decl = FnDecl { inputs: vec![], output };
+
+        let task_context_id = self.resolver.next_node_id();
+        let task_context_hid = self.lower_node_id(task_context_id);
+        let ast_decl = FnDecl {
```

Comment:
Style: lowering code is typically written using more `let` bindings to break up large expressions like this.